### ISSUE TITLE
Add COVID-19 SpecialAnnouncement markup for Google results?

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-sitebanner#12]
This module has a fairly large conceptual overlap with a recently announced feature by Google: [COVID-19 Special Announcements](https://webmasters.googleblog.com/2020/04/highlight-covid-19-announcements-search.html). The banner module aims to present one or more "global" banners as unstructured content on the website itself. The Google feature embeds structured metadata in the website, which then gets picked up by Google for presentation in the search result.

Similarities:
- Both can have multiple banners on a page
- Both have required content (Google's `text` property)
- Both can have expiry dates

Differences:
- Google announcements require a title (`name` property)
- Google announcements are time limited (max 1 month)
- Google announcements text HTML is more limited (e.g. no `<h1-5>`), and is likely more restricted in character counts. Those [restrictions](https://developers.google.com/search/docs/data-types/special-announcements#data-type-definitions) probably make sense for both cases though
- Google announcements can optionally have spatial coverage. While we could implement that as site banners through GeoIP lookups, it would complicate the module
- Google announcements can optionally have "announcement locations" (e.g. school closures, testing locations)
- Google announcements should be restricted to the pages they're relevant to. It's unclear whether "global" announcements are allowed or encouraged.
- Google announcements has all kinds of additional metadata (mainly URLs)
- Google announcements have more validation constraints (e.g. either `text` or a URL property)
- There's no in-built preview, although we could use `shareddraftcontent` to [hotlink](https://search.google.com/test/rich-results?url=http%3A%2F%2Fsilverstripe.org%2F) to Google's Rich Results Test
- "publication" of a banner relies on recrawls (could take days, based on Googlebot heuristics), or more sophisticated Google Sitemap usage than most sites have right now (https://github.com/wilr/silverstripe-googlesitemaps with `google_notification_enabled=true`)
- Site banners support a "alert type" (e.g. info vs. warning)

Notes:
 * I think it makes sense to make banners optionally page-specific (with inheritance). It adds more queries to each page view, but those should be reasonably cacheable.
 * We could standardise to create banners out of the structured data on-site as well, although that'll complicate the developer experience.
 * There's probably an opportunity to use an embedded RDF editor rather than modelling this all through database columns, potentially even with built-in UX for selecting places, client-side validation based on the schema data, etc

Given these complexities, any module behaviour around Google would need to be opt-in (by developer, then by author on a specific banner instance). Any location specific metadata would be discarded in the on-site banners (too hard). We'd need to create more complicated on-site views to account for all the other metadata (mainly links), or discard that as well. And we'd need to add support for title/content combos in on-site views.

In conclusion, this isn't a natural fit. I'm tending towards making this a standalone module. Integration into this module makes sense if we assume that Google retains this functionality beyond the COVID19 use case, and that website owners and users actually find it useful (with lots of incoming traffic through Google search).
